runtime/src/iree-amd-aie/aie_runtime/iree_aie_runtime.cc

```diff
@@ -629,6 +629,135 @@ struct AMDAIEDeviceModel getDeviceModel(AMDAIEDevice device) {
   llvm::report_fatal_error("Unhandled AMDAIEDevice case");
 }
 
+/// Generate a DenseMap key we can use for the element types (alternatives
+/// considered: implement tombstone for std::array, or use std::map instead of
+/// DenseMap).
+static constexpr uint32_t getElementTypeKey(uint32_t a, uint32_t b,
+                                            uint32_t c) {
+  return a + (b << 8) + (c << 16);
+}
+
+/// Map from (LHS bitwidth, RHS bitwidth, Accumulator bitwidth) to the AIE
+/// instruction size (m, n, k) for the integer types with those bitwidths.
+/// This function is based on the following table pulled from the
+/// AIEVec_MatMulOp documentation in
+/// mlir-aie/include/aie/Dialect/AIEVec/IR/AIEVecOps.td
+///
+///   lhs                | rhs                | accumulator
+///  :------------------:|:------------------:|:-----------------:
+///   `vector<4x16xi8>`  | `vector<16x8xi4>`  | `vector<4x8xi32>`
+///   `vector<4x8xi8>`   | `vector<8x8xi8>`   | `vector<4x8xi32>`
+///   `vector<4x4xi16>`  | `vector<4x8xi8>`   | `vector<4x8xi32>`
+///   `vector<4x2xi16>`  | `vector<2x8xi16>`  | `vector<4x8xi32>`
+///   `vector<2x8xi16>`  | `vector<8x8xi8>`   | `vector<2x8xi64>`
+///   `vector<4x8xi16>`  | `vector<8x4xi8>`   | `vector<4x4xi64>`
+///   `vector<2x4xi16>`  | `vector<4x8xi16>`  | `vector<2x8xi64>`
+///   `vector<4x4xi16>`  | `vector<4x4xi16>`  | `vector<4x4xi64>`
+///   `vector<4x2xi32>`  | `vector<2x4xi16>`  | `vector<4x4xi64>`
+///   `vector<4x8xbf16>` | `vector<8x4xbf16>` | `vector<4x4xf32>`
+///
+/// An instruction size (m, n, k) is returned for each combination of element
+/// type in the table. Combinations of element type that are not covered by the
+/// table return failure.
+///
+/// Example: consider the first line of the table:
+///   `vector<4x16xi8>`  | `vector<16x8xi4>`  | `vector<4x8xi32>`
+///
+/// This first line says that if 'lhs' is an i8 tensor, 'rhs' is an i4 tensor
+/// and 'accumulator' is an i32 tensor, then there is an AIE instruction for
+/// matmul with m = 4, n = 8, k = 16.
+static llvm::DenseMap<uint32_t, std::array<uint32_t, 3>>
+    &getIntegerMatmulInstructionSizeMap() {
```

Comment:
This would be a good time to rename this to `getNpu1IntegerMatmulInstructionSizeMap`

Reply:
I don't see this having been renamed

Reply:
Oops. My bad. Did it now.